plugins/kubernetes/app/models/kubernetes/deploy_executor.rb

```diff
@@ -85,29 +85,45 @@ def fetch_pods(client, query)
     end
 
     def show_failure_cause(release)
-      bad_pods = release.clients.flat_map do |client, query, deploy_group|
-        bad_pods = fetch_pods(client, query).select { |p| p.restarted? || !p.live? }
-        bad_pods.map { |p| [p, client, deploy_group] }
-      end
-
-      bad_pods.each do |pod, client, deploy_group|
-        namespace = deploy_group.kubernetes_namespace
+      bad_pods(release).each do |pod, client, deploy_group|
         @output.puts "\n#{deploy_group.name} pod #{pod.name}:"
+        namespace = deploy_group.kubernetes_namespace
```

Comment:
FYI, if you're getting the pod back from Kubernetes, the namespace is in the pod itself.  `pod.metadata.namespace`, i believe.